tests/jobserver/api/test_releases.py

```diff
@@ -187,3 +188,74 @@ def test_upload_api_redirected(api_rf):
     release.refresh_from_db()
     assert response["Release-Id"] == release.id
     assert response["Location"] == f"http://testserver{release.get_absolute_url()}"
+
+
+@pytest.mark.django_db
+def test_release_index_api(api_rf):
+    release = ReleaseFactory()
+    user = UserFactory()
+    client = APIClient()
+
+    # bad id
+    response = client.get("/api/v2/releases/badid")
+    assert response.status_code == 404
+
+    index_url = f"/api/v2/releases/{release.id}"
+
+    # not logged in
+    response = client.get(index_url)
+    assert response.status_code == 403
+
+    # logged in, but no permission
+    client.force_authenticate(user=user)
+    response = client.get(index_url)
+    assert response.status_code == 403
+
+    # logged in, with permission
+    ProjectMembershipFactory(user=user, project=release.workspace.project)
+    response = client.get(index_url)
+    assert response.status_code == 200
+    assert response.json() == ["file.txt", "metadata/manifest.json"]
+
+
+@pytest.mark.django_db
+def test_release_file_api(api_rf):
+    release = ReleaseFactory()
+    user = UserFactory()
+    client = APIClient()
+
+    response = client.get("/api/v2/releases/badid/file.txt")
+    assert response.status_code == 404
+
+    file_url = f"/api/v2/releases/{release.id}/file.txt"
+
+    # not logged in
+    response = client.get(file_url)
+    assert response.status_code == 403
+
+    # logged in, but no permission
+    client.force_authenticate(user=user)
+    response = client.get(file_url)
+    assert response.status_code == 403
+
+    # logged in, with permission
+    ProjectMembershipFactory(user=user, project=release.workspace.project)
+    response = client.get(file_url)
+    assert response.status_code == 200
+    assert response.content == b"test"
+    assert response.headers["Content-Type"] == "text/plain"
+    assert response.headers["Content-Dispostion"] == 'attachment; filename="file.txt"'
+
+    # test nginx configuration
+    response = client.get(file_url, HTTP_RELEASES_REDIRECT="/releases")
+    assert response.status_code == 200
+    assert (
+        response.headers["X-Accel-Redirect"]
+        == f"/releases/{release.workspace.name}/{release.id}/file.txt"
+    )
+    assert response.headers["Content-Dispostion"] == 'attachment; filename="file.txt"'
+
+    # delete file
+    release.file_path("file.txt").unlink()
+    response = client.get(file_url)
+    assert response.status_code == 404
```

Comment:
I'd prefer to break these down into separate unit tests if possible, so if say we break the nginx config that test breaks rather than the tests not using it.

Reply:
Understandable. I prefer this lighter-weight approach in general, but will refactor